--- clair-scanner.orig/docker.go	2021-12-17 09:34:57.465975718 +0800
+++ clair-scanner/docker.go	2021-12-17 09:41:27.809360196 +0800
@@ -1,10 +1,10 @@
 package main
 
 import (
-	"context"
 	"encoding/json"
 	"io"
 	"os"
+	"os/exec"
 	"strings"
 
 	"github.com/docker/docker/client"
@@ -18,17 +18,23 @@
 
 // saveDockerImage saves Docker image to temorary folder
 func saveDockerImage(imageName string, tmpPath string) {
-	docker := createDockerClient()
-
-	imageReader, err := docker.ImageSave(context.Background(), []string{imageName})
-	if err != nil {
-		logger.Fatalf("Could not save Docker image [%s]: %v", imageName, err)
-	}
-
-	defer imageReader.Close()
-
-	if err = untar(imageReader, tmpPath); err != nil {
-		logger.Fatalf("Could not save Docker image: could not untar [%s]: %v", imageName, err)
+	
+	reader, writer := io.Pipe()
+	defer writer.Close()
+	
+	cmd := exec.Command("bash","-c", "podman save " + imageName)
+	cmd.Stdout = writer
+
+	go func() {
+		defer reader.Close()
+
+		if err := untar(reader, tmpPath); err != nil {
+			logger.Fatalf("Could not save Docker image: could not untar [%s]: %v", imageName, err)
+		}
+	}()
+	
+	if err := cmd.Run(); err != nil {
+		logger.Fatalf("Could not run the podman command: %s\n", err)
 	}
 }
 
